vdr/vdr.go

```diff
@@ -69,6 +70,43 @@ type VDR struct {
 	eventManager      events.Event
 }
 
+func (r *VDR) DeriveWebDIDDocument(ctx context.Context, baseURL url.URL, nutsDID did.DID) (*did.Document, error) {
+	nutsDIDDocument, _, err := r.Resolver().Resolve(nutsDID, nil)
+	if err != nil {
+		return nil, err
+	}
+	isOwner, err := r.IsOwner(ctx, nutsDID)
+	if err != nil {
+		return nil, err
+	}
+	if !isOwner {
+		log.Logger().
+			WithError(err).
+			WithField(core.LogFieldDID, nutsDID).
+			Info("Tried to derive did:web document from Nuts DID that is not owned by this node")
+		return nil, types.ErrNotFound
+	}
+
+	resultDIDDocumentData, _ := nutsDIDDocument.MarshalJSON()
+	// Replace did:nuts DIDs with did:web, but only for owned DIDs
+	webDID, err := didweb.URLToDID(*baseURL.JoinPath(nutsDID.ID))
+	if err != nil {
+		return nil, fmt.Errorf("unable to derive Web DID from Nuts DID (%s): %w", nutsDID, err)
+	}
+	resultDIDDocumentData = bytes.ReplaceAll(resultDIDDocumentData, []byte(nutsDID.String()), []byte(webDID.String()))
+	var result did.Document
+	if err = result.UnmarshalJSON(resultDIDDocumentData); err != nil {
+		return nil, fmt.Errorf("did:web unmarshal error (%s): %w", nutsDID, err)
+	}
+	result.AlsoKnownAs = append(result.AlsoKnownAs, nutsDID.URI())
+	// did:web support is currently just for supporting third party systems resolving key material,
+	// so no need to retain services (which often refer to services in other did:nuts documents, complicating things).
+	result.Service = nil
```

Comment:
Should this have a TODO with an issue?

Reply:
I don't think so, since it's currently not a feature we plan to have? So not TODO per se?